### PR TITLE
Implement shared waitfor/notify across threads

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: src/modules/lua_mtev.c [hunk 1]
```diff
@@ -5286,10 +5293,208 @@ mtev_lua_deserialize(lua_State *L, const lua_data_t *data){
...
+  }
+
+  /* There is no item yes, so we need to wait for one assuming timeout is non-zero */
+  if(timeout == 0) return 0;
+
+  mtev_lua_resume_info_t *ci;
+  ci = mtev_lua_find_resume_info(L, mtev_true); /* we cannot suspect without context */
+
+  struct shared_queues_wait *crutch = calloc(1, sizeof(*crutch));
+  double ntimeout = timeout < 0 ? 10900 : timeout;
```
Comment (cody271) [inline src/modules/lua_mtev.c]: Where does the value `10900` here come from?

File: src/modules/lua_mtev.c [hunk 2]
```diff
@@ -5286,10 +5293,208 @@ mtev_lua_deserialize(lua_State *L, const lua_data_t *data){
...
+      mtev_lua_free_data((lua_data_t *)node->data);
+    }
+    else {
+      lua_pushnil(L);
+    }
+    free(node);
+    return 1;
+  }
+
+  /* There is no item yes, so we need to wait for one assuming timeout is non-zero */
```
Comment (cody271) [inline src/modules/lua_mtev.c]:
```suggestion
  /* There is no item yet, so we need to wait for one assuming timeout is non-zero */
```

File: src/modules/lua_mtev.c [hunk 3]
```diff
@@ -5286,10 +5293,208 @@ mtev_lua_deserialize(lua_State *L, const lua_data_t *data){
...
+  if(timeout == 0) return 0;
+
+  mtev_lua_resume_info_t *ci;
+  ci = mtev_lua_find_resume_info(L, mtev_true); /* we cannot suspect without context */
```
Comment (cody271) [inline src/modules/lua_mtev.c]:
```suggestion
  ci = mtev_lua_find_resume_info(L, mtev_true); /* we cannot suspend without context */
```

File: src/modules/lua_mtev.c (hunk 1 above)
Comment (postwait) [reply src/modules/lua_mtev.c]: Ha. Actually a typo, supposed to be 10000. But it is just an arbitrarily large timeout.

Comment (postwait) [reply src/modules/lua_mtev.c]: resolved in comment.

**Pull request merged**